elastic/kibana
# [Index Management] Add an index details page behind a dev feature flag

## Summary
This PR adds a dev feature flag `xpack.index_management.dev.enableIndexDetailsPage` that will allow us to build out the new index details page in small iterations. Without the flag, the UI of Index Management is not changed. A skeleton component is created for the details page (see screenshot below). 

### How to test
1. Test the Index Management UI (Indices tab) without the flag and check that no changes were introduced
1. Add `xpack.index_management.dev.enableIndexDetailsPage: true` to the file `/config/kibana.dev.yml`
2. Navigate to the Indices tab in Index Management, toggle "hidden indices" if no indices exist and click any index name
3. Check that the new index details page is displayed
4. Check that the tabs on the page are working

### Screenshots 
<img width="1209" alt="Screenshot 2023-08-09 at 19 17 46" src="https://github.com/elastic/kibana/assets/6585477/e654ef36-ccf3-40a4-8c7b-750b83defef5">

### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
